metaflow/plugins/deployer.py

```diff
@@ -0,0 +1,192 @@
+    if _type is None:
+        raise ValueError(
+            "Deployer doesn't have a 'type' to target. Please use a sub-class of Deployer."
+        )
+    if name is None:
+        return getattr(api(**top_level_kwargs), _type)()
+    return getattr(api(**top_level_kwargs), _type)(name=name)
+
+
+class TriggeredRun(object):
+    def __init__(
+        self,
+        deployer: "Deployer",
+        content: str,
+    ):
+        self.deployer = deployer
+        content_json = json.loads(content)
+        self.metadata_for_flow = content_json.get("metadata")
+        self.pathspec = content_json.get("pathspec")
+        self.deployer.name = content_json.get("name")
+
+    @property
+    def name(self):
+        return self.deployer.name
+
+    @property
+    def status(self):
+        raise NotImplementedError
+
+    @property
+    def run(self):
+        clear_and_set_os_environ(self.deployer.old_env)
+        metadata(self.metadata_for_flow)
+        try:
+            return Run(self.pathspec, _namespace_check=False)
+        except MetaflowNotFound:
+            return None
+
+    def terminate(self, **kwargs):
+        _, run_id = self.pathspec.split("/")
+        command = get_lower_level_group(
+            self.deployer.api,
+            self.deployer.top_level_kwargs,
+            self.deployer.type,
+            self.deployer.name,
+        ).terminate(run_id=run_id, **kwargs)
+
+        pid = self.deployer.spm.run_command(
+            [sys.executable, *command],
+            env=self.deployer.env_vars,
+            cwd=self.deployer.cwd,
+            show_output=self.deployer.show_output,
+        )
+
+        command_obj = self.deployer.spm.get(pid)
+        return command_obj.process.returncode == 0
+
+
+class DeployedFlow(object):
+    def __init__(
+        self,
+        deployer: "Deployer",
+    ):
+        self.deployer = deployer
+
+    @property
+    def production_token(self):
+        raise NotImplementedError
+
+    def trigger(self, **kwargs) -> "TriggeredRun":
+        return self.deployer.trigger(**kwargs)
+
+
+class Deployer(object):
+    type: ClassVar[Optional[str]] = None
+
+    def __init__(
+        self,
+        flow_file: str,
+        name: Optional[str] = None,
+        show_output: bool = False,
+        profile: Optional[str] = None,
+        env: Optional[Dict] = None,
+        cwd: Optional[str] = None,
+        **kwargs
+    ):
+        from metaflow.cli import start
+        from metaflow.runner.click_api import MetaflowAPI
+
+        self.flow_file = flow_file
+
+        self.name = name
+        self.show_output = show_output
+
+        self.old_env = os.environ.copy()
+        self.env_vars = self.old_env.copy()
+        self.env_vars.update(env or {})
+        if profile:
+            self.env_vars["METAFLOW_PROFILE"] = profile
+
+        self.cwd = cwd
+        self.spm = SubprocessManager()
+        self.top_level_kwargs = kwargs
```

Comment:
There seems to be one level missing. We can have things like: `myflow.py <tl options> argo/sfn/whatever <argo/sfn/whatever options> create <create options>`. I only see <tl options> and <create options> here I think.